### PR TITLE
Plane construction and extraction.

### PR DESCRIPTION
**Pull request by lispandfound**
### Enhancements to `Plane` class:
* Added new properties `strike`, `strike_nztm`, `dip_dir`, and `dip_dir_nztm` to the `Plane` class to handle different coordinate systems and bearings. (`source_modelling/sources.py`) [[1]](diffhunk://#diff-5f2630533f745c6079637bcfdab384f3d08c41a5b14c36173b4dbc55423dc984R291-R297) [[2]](diffhunk://#diff-5f2630533f745c6079637bcfdab384f3d08c41a5b14c36173b4dbc55423dc984R308-R314)
* Refactored the `from_centroid_strike_dip` method to include additional parameters such as `strike_nztm` and `dip_dir_nztm` and added comprehensive validation checks for input consistency. (`source_modelling/sources.py`) [[1]](diffhunk://#diff-5f2630533f745c6079637bcfdab384f3d08c41a5b14c36173b4dbc55423dc984L326-R348) [[2]](diffhunk://#diff-5f2630533f745c6079637bcfdab384f3d08c41a5b14c36173b4dbc55423dc984L342-R477)

### SRF Module:
* Added a new property `planes` to the `segments` method to extract planes from SRF segments. (`source_modelling/srf.py`)

### Test Coverage:

* Added new test cases to validate the construction of `Plane` objects, including failure cases for invalid input combinations and parameter derivations. (`tests/test_sources.py`) [[1]](diffhunk://#diff-3cd5d0943d234b07cb43ae8d4ddb6ac1030183923dc4d6cd56e528f646e48ecdR4) [[2]](diffhunk://#diff-3cd5d0943d234b07cb43ae8d4ddb6ac1030183923dc4d6cd56e528f646e48ecdL206-R392) [[3]](diffhunk://#diff-3cd5d0943d234b07cb43ae8d4ddb6ac1030183923dc4d6cd56e528f646e48ecdL299-R426)
* Updated existing test cases to reflect the changes in the `Plane` class and its methods. (`tests/test_sources.py`) [[1]](diffhunk://#diff-3cd5d0943d234b07cb43ae8d4ddb6ac1030183923dc4d6cd56e528f646e48ecdL326-R448) [[2]](diffhunk://#diff-3cd5d0943d234b07cb43ae8d4ddb6ac1030183923dc4d6cd56e528f646e48ecdL353-R459) [[3]](diffhunk://#diff-3cd5d0943d234b07cb43ae8d4ddb6ac1030183923dc4d6cd56e528f646e48ecdL477)
* Added assertions to verify the consistency of SRF header data with the newly introduced `planes` property. (`tests/test_srf.py`) [[1]](diffhunk://#diff-931748e37d1fdd2a8b0e2329f848e06a857c9df6cc8f5475138abf9b2af6134eR85-R94) [[2]](diffhunk://#diff-931748e37d1fdd2a8b0e2329f848e06a857c9df6cc8f5475138abf9b2af6134eR203-R211)

### CODE REVIEW

File: source_modelling/sources.py [hunk 1]
```diff
@@ -339,31 +356,125 @@ def from_centroid_strike_dip(
...
-        strike : float
-            The strike of the fault (in degrees).
-        dip_dir : Optional[float]
-            The dip direction of the fault (in degrees). If None this is assumed to be strike + 90 degrees.
-        top : float
-            The top depth of the plane (in km).
-        bottom : float
-            The bottom depth of the plane (in km).
+            The centre of the fault plane in lat, lon, and optionally depth (km) coordinates.
+        dip : float
```
Comment (joelridden) [inline source_modelling/sources.py:360]: Maybe just for extra clarity could add WGS84 bearing for dip and strike

File: tests/test_srf.py [hunk 2]
```diff
@@ -190,6 +200,15 @@ def test_darfield_srf():
         assert len(segment) == segment_header["nstk"] * segment_header["ndip"]
         assert (segment["dip"] == segment_header["dip"]).all()
         assert (segment["stk"] == segment_header["stk"]).all()
+    for (_, header), plane in zip(darfield_srf.header.iterrows(), darfield_srf.planes):
+        assert header[["elat", "elon"]].values == pytest.approx(plane.centroid[:2])
+        assert header["wid"] == pytest.approx(plane.width)
+        assert header["len"] == pytest.approx(plane.length)
+        assert header["len"] == pytest.approx(plane.length)
+        # the dip in the Darfield SRF makes no sense!
+        # assert header["dip"] == pytest.approx(plane.dip)
```
Comment (joelridden) [inline tests/test_srf.py:209]: Is this fine?

File: source_modelling/sources.py (hunk 1 above)
Comment (lispandfound) [reply source_modelling/sources.py:360]: dip is not a bearing from north so it is not a WGS84 bearing. Strike and dip direction sure.

File: tests/test_srf.py (hunk 2 above)
Comment (lispandfound) [reply tests/test_srf.py:209]: The "dip" given in the SRF is > 90 degrees which makes no sense.

File: source_modelling/sources.py (hunk 1 above)
Comment (joelridden) [reply source_modelling/sources.py:360]: Sorry meant dip_dir not dip

**Review by joelridden — APPROVED** [review]

**Review by claudio525 — APPROVED** [review]

**Pull request merged**